src/c/src/chat_history.cpp

```diff
@@ -0,0 +1,371 @@
+    if (!history) {
+        return OV_GENAI_CHAT_HISTORY_INVALID_PARAM;
+    }
+    try {
+        std::unique_ptr<ov_genai_chat_history> _history = std::make_unique<ov_genai_chat_history>();
+        _history->object = std::make_shared<ov::genai::ChatHistory>();
+        *history = _history.release();
+    } catch (const std::exception& e) {
+        return OV_GENAI_CHAT_HISTORY_ERROR;
+    } catch (...) {
+        return OV_GENAI_CHAT_HISTORY_ERROR;
+    }
+    return OV_GENAI_CHAT_HISTORY_OK;
+}
+
+ov_genai_chat_history_status_e ov_genai_chat_history_create_from_json(
+    const char* messages_json,
+    ov_genai_chat_history** history) {
+    if (!messages_json || !history) {
+        return OV_GENAI_CHAT_HISTORY_INVALID_PARAM;
+    }
+    try {
+        ov::genai::JsonContainer messages = ov::genai::JsonContainer::from_json_string(std::string(messages_json));
+        if (!messages.is_array()) {
+            return OV_GENAI_CHAT_HISTORY_INVALID_JSON;
+        }
+        std::unique_ptr<ov_genai_chat_history> _history = std::make_unique<ov_genai_chat_history>();
+        _history->object = std::make_shared<ov::genai::ChatHistory>(messages);
+        *history = _history.release();
+    } catch (const ov::Exception& e) {
+        return OV_GENAI_CHAT_HISTORY_INVALID_JSON;
+    } catch (const std::exception& e) {
+        return OV_GENAI_CHAT_HISTORY_ERROR;
+    } catch (...) {
+        return OV_GENAI_CHAT_HISTORY_ERROR;
+    }
+    return OV_GENAI_CHAT_HISTORY_OK;
+}
+
+void ov_genai_chat_history_free(ov_genai_chat_history* history) {
+    if (history) {
+        delete history;
+    }
+}
+
+ov_genai_chat_history_status_e ov_genai_chat_history_push_back(
+    ov_genai_chat_history* history,
+    const char* message_json) {
+    if (!history || !(history->object) || !message_json) {
+        return OV_GENAI_CHAT_HISTORY_INVALID_PARAM;
+    }
+    try {
+        ov::genai::JsonContainer message = ov::genai::JsonContainer::from_json_string(std::string(message_json));
+        history->object->push_back(message);
+    } catch (const ov::Exception& e) {
+        return OV_GENAI_CHAT_HISTORY_INVALID_JSON;
+    } catch (const std::exception& e) {
+        return OV_GENAI_CHAT_HISTORY_ERROR;
+    } catch (...) {
+        return OV_GENAI_CHAT_HISTORY_ERROR;
+    }
+    return OV_GENAI_CHAT_HISTORY_OK;
+}
+
+ov_genai_chat_history_status_e ov_genai_chat_history_pop_back(ov_genai_chat_history* history) {
+    if (!history || !(history->object)) {
+        return OV_GENAI_CHAT_HISTORY_INVALID_PARAM;
+    }
+    try {
+        if (history->object->empty()) {
+            return OV_GENAI_CHAT_HISTORY_EMPTY;
+        }
+        history->object->pop_back();
+    } catch (const ov::Exception& e) {
+        return OV_GENAI_CHAT_HISTORY_EMPTY;
+    } catch (const std::exception& e) {
+        return OV_GENAI_CHAT_HISTORY_ERROR;
+    } catch (...) {
+        return OV_GENAI_CHAT_HISTORY_ERROR;
+    }
```

Comment:
Why different catch blocks are needed if all of them return the same error status?